janusgraph-driver/src/main/java/org/janusgraph/util/IDUtils.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2022 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.util;
+
+public class IDUtils {
+    public static void checkVertexId(Object vertexId) {
+        if (vertexId == null) {
+            throw new IllegalArgumentException("vertex id cannot be null");
+        }
+        if (vertexId instanceof String) {
+            // string-type vertex id is allowed
+            return;
+        }
+        if (vertexId instanceof Number) {
+            if (((Number) vertexId).longValue() <= 0) {
+                throw new IllegalArgumentException(String.format("vertex id %d is non-positive", ((Number) vertexId).longValue()));
```

Comment:
Not related to this PR but do you know by any chance why negative vertex ids are not allowed?

Reply:
Iff `ids.flush=false`, then JanusGraph will assign a temporary negative ID when an element is created and generates a unique positive ID only when the transaction commits. Thus, a negative number represents a temporary id.

Reply:
Thank you, got it! I guess it would make sense to think about reducing that scope (I guess `2^31` would be enough for temporary ids and the rest could be potentially used for normal ids) but it's definitely out of the scope of this PR.